dec-no-dos/dec.h

```diff
@@ -0,0 +1,129 @@
+/** \file		dec.h
+	\brief		File performing decryption
+	\details	The decryption is based on the fact that the key used in XOR encryption with ECB mode, will be easily spotable over a string of bytes of 0. 
+	\author		Monika Olchowik
+	\date		12.2016
+	\version	1.0
+*/
+
+#ifndef DEC
+#define DEC
+
+#include<algorithm>
+#include<cstring>
+#include<cstdio>
+
+
+#ifndef BUFF_SIZE
+// Maksymalna wielkosc bufora
+#ifdef BIG_BUFF				/**< Big buffor size was only used for testing purposes only */
+#define BUFF_SIZE 524288	/**< Buffer size */
+#define OBR 524288			/**< The number of bytes in which we are looking for the key */
+#else
+#define BUFF_SIZE 1096
+#define OBR 1024
+#endif
+#endif
+
+#ifndef MAX_KEY_SIZE
+#define MAX_KEY_SIZE 128	/**< Maximum viable key size */
+#endif
+
+#ifndef MAXN
+#define MAXN 524288			/**< Variable used for the Hashmap */
+#endif
+
+#ifndef MAXP
+#define MAXP 185531			/**< Variable used for the Hashmap */
+#endif
+
+#define EPSILON_COMPARE 0.0001
+
+#include"vali.h"
+
+/**
+	/brief	Structer used for the hash of 
+*/
+struct hash_table
```

Comment:
It is not a table. It is an entry in some table, but name does not reflect that.